File: docs/samples/getting-started/sample.mjs
import { DataHubClient } from "@nfdi4plants/datahub-client";
import { Authentication } from "@nfdi4plants/datahub-client/Http/Authentication.js";

const url = "https://git.nfdi4plants.org";

// PersonalAccessToken, OAuthToken, and JobToken cover the GitLab credential types.
const auth = Authentication.PersonalAccessToken("your-datahub-pat");

// The client defaults to the HTTP transport built for the host runtime;
// assign `client.Http` to substitute your own (retries, proxies, tests).
const client = new DataHubClient(url, auth);

console.log(`DataHubClient ready against ${url}`);

File: docs/samples/getting-started/sample.py
from datahub_client import DataHubClient, Authentication

url = "https://git.nfdi4plants.org"

# PersonalAccessToken, OAuthToken, and JobToken cover the GitLab credential types.
auth = Authentication.PersonalAccessToken("your-datahub-pat")

# The client defaults to the HTTP transport built for the host runtime;
# assign `client.Http` to substitute your own (retries, proxies, tests).
client = DataHubClient(url, auth)

print(f"DataHubClient ready against {url}")
